hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/catalog/TestHoodieHiveCatalog.java

```diff
@@ -270,6 +271,20 @@ public void testCreateNonHoodieTable() throws TableAlreadyExistException, Databa
     }
   }
 
+  @Test
+  public void testCreateHoodieTableWithWrongTableType() throws TableAlreadyExistException, DatabaseNotExistException {
+    HashMap<String,String> properties = new HashMap<>();
+    properties.put(FactoryUtil.CONNECTOR.key(), "hudi-fake");
+    properties.put("table.type","wrong type");
+    CatalogTable table =
+            new CatalogTableImpl(schema,  properties, "hudi table");
+    try {
+      hoodieCatalog.createTable(tablePath, table, false);
+    } catch (HoodieValidationException e) {
+      assertEquals("table.type's value is invalid", e.getMessage());
```

Comment:
This patch fixes the hoodie catalog instead of hoodie hive catalog right?

Reply:
this patch fix the hoodie hive catalog, when using hudi hive catalog ,using create if not exists table , the code will go to org.apache.hudi.table.catalog.HoodieHiveCatalog#createTable | org.apache.hudi.table.catalog.HoodieHiveCatalog#instantiateHiveTable| org.apache.hudi.table.catalog.TableOptionProperties#translateFlinkTableProperties2Spark 
and in this method if the table.type is wrong it will throw npe

Reply:
Can we use assertThrows instead?